### PR TITLE
Significant changes and corrections to the SE health doc

### LINKED ISSUE #7945: 4.x SE health page and guide need to refer to the new required dependency and update the endpoints in a couple places
## Environment Details
* Helidon Version: 4.0.0
* Helidon SE or Helidon MP SE
* JDK version:
* OS:
* Docker version (if applicable):

----------

## Problem Description
The SE health guide and pages should mention the new dependency on `helidon-webserver-observe-health`.

Most examples were corrected earlier, but there are a few places where the old endpoints are still used in examples.

### PR DESCRIPTION
**Pull request by tjquinno**
### Description
Resolves #7945 

The health doc was significantly out of step with the late coding changes for 4.0.0. This PR addresses a number of problems, including what config key prefix is correct and the code examples to create a custom observe feature and a custom health observer in which to add custom health checks.

### Documentation
This is a doc PR.

### CODE REVIEW

File: docs/se/guides/health.adoc [hunk 1]
```diff
@@ -158,96 +165,129 @@ curl http://localhost:8080/observe/health
...
+                             .addCheck(() -> HealthCheckResponse.builder() // <5>
+                                               .status(System.currentTimeMillis() - serverStartTime.get() >= 8000)
+                                               .detail("time", System.currentTimeMillis())
+                                               .build(),
+                                       HealthCheckType.STARTUP,
+                                       "warmedUp")
+                             .build()) // <6>
+        .build();
+
+    WebServer server = WebServer.builder()
```
Comment (spericas) [inline docs/se/guides/health.adoc:200]: Extra indentation here, right?

File: docs/se/guides/health.adoc [hunk 2]
```diff
@@ -271,28 +311,37 @@ curl http://localhost:8080/observe/started
...
+=== Applying Configuration to a Custom Health Observer: Customizing the URL path
+Earlier examples showed how to add custom health checks by building a custom `HealthObserver` in which the code set up the behavior of the health subsystem explicitly. Recall that the example code invoked the `HealthObserver.Builder` `details` method to turn on detailed output.
+
+Once it creates a custom health observer, your code has full responsibility for determining the observer's behavior; Helidon does not automatically apply configuration to a custom observer. But your code can easily do so.
 
-You can use a custom URL path for heath checks by setting the `endpoint` for the `HealthObserver`.
+The next example customizes the URL path for the health endpoint, first explicitly in the code and then via configuration.
+
+==== Customizing the endpoint path in the code
+Customize the URL path for heath checks by invoking the `endpoint` method on the `HealthObserver.Builder`.
```
Comment (spericas) [inline docs/se/guides/health.adoc:332]: "heath" -> "health"

File: docs/se/guides/health.adoc [hunk 3]
```diff
@@ -303,6 +352,40 @@ ObserveFeature observe = ObserveFeature.builder()
 curl http://localhost:8080/myhealth
 ----
 
+Earlier you added health config to the `application.yaml` config file to turn on detailed output. If you want to run an experiment, change that `details` setting in the config file to `false` and stop, rebuild, and rerun the application. Now access the health endpoint (at `/myhealth`, remember).
+The output _remains_ detailed because your code--which has full responsibility for determining the custom health observer's behavior--does not apply configuration to the custom observer's builder.
+
+==== Adding configuration to a custom observer
+In addition to preparing the health observer builder with hard-coded settings, your code can also apply configuration for health so your user could set any health behavior there as well.
+
```
Comment (spericas) [inline docs/se/guides/health.adoc:360]: Was confused by "your user" here. Maybe rephrase this? Maybe "your code can also apply configuration by directly referring to a config node" or something like that.

File: docs/se/guides/health.adoc [hunk 4]
```diff
@@ -303,6 +352,40 @@ ObserveFeature observe = ObserveFeature.builder()
...
+                                       "warmedUp")
+                             .config(config.get("server.features.observe.observers.health")) // <1>
+                             .build()
+        )
+        .build();
+----
+<1> Find and apply any health-related settings from configuration at the `server.features.observe.observers.health` config key.
+
+Your code decides what config key to use for retrieving the configuration. Recall earlier, before adding custom health checks, you added a config section for health--to set `details` to `true`--at `server.features.observe.observers.health`. Helidon used that configuration to set up the health observer _it_ created automatically. To be consistent for your users, it's a good idea for your application code, as it preares a custom `HealthObserver`, to look in the same place Helidon does for health config.
+
```
Comment (spericas) [inline docs/se/guides/health.adoc:385]: "preares" -> "prepares"?

File: docs/se/guides/health.adoc [hunk 5]
```diff
@@ -303,6 +352,40 @@ ObserveFeature observe = ObserveFeature.builder()
...
+Order is important. Here, the code first sets `details` to `true` explicitly and later applies configuration. If your end user sets `details` in the `server.features.observe.observers.health` config to `false`, that setting overrides the hard-coded `true` setting in the code _because of where in the code you apply the configuration_. Try changing the `details` value to `false` in the config file and then stop, rebuild, and rerun the application. Access the health endpoint and notice that the output is no longer detailed.
+
+In general, most applications should apply settings from config _after_ assigning any settings in the code so users have the final say, but there might be exceptions in your particular case.
```
Comment (spericas) [inline docs/se/guides/health.adoc:388]: I see the "users" again. I'd would use "external config" or simply "config" instead, but I understand the notion better now.

File: docs/se/health.adoc [hunk 6]
```diff
@@ -76,46 +77,161 @@ Optional dependency to use built-in health checks:
...
-* use a method reference a method with the double colon operator `::`.
+* Pass the name and type of the health check and a `Supplier` of a `HealthCheckResponse` such as a method reference or a lambda expression.
+* Pass an instance of a class which implements the `HealthCheck` interface.
+
+Within an application different techniques might make sense for different custom health checks, depending on the complexity of the logic for computing the status for each check.
+The various styles are functionally equivalent; for a given custom health check choose the style which enhances the readability and clarity of your code.
+The examples below, in no particular order, implement the same custom health check functionality in different ways to illustrate.
+
+==== Option 1: Using a `HealthCheckResponse` supplier method
+If you gather the logic for computing the health check response into a method you can use a method reference to register the health check.
```
Comment (spericas) [inline docs/se/health.adoc:111]: "method you" -> "method, you"?

File: docs/se/health.adoc [hunk 7]
```diff
@@ -76,46 +77,161 @@ Optional dependency to use built-in health checks:
...
+[source,java]
+.Registering a health check using a method reference
+----
+ObserveFeature observe = ObserveFeature.builder()
+                             .config(config.get("server.features.observe")) // <1>
+                             .addObserver(HealthObserver.builder()          // <2>
+                             .useSystemServices(true) // Include Helidon-provided health checks.
+                             .addCheck(Main::slowStartLivenessResponse, // <3>
+                                       HealthCheckType.READINESS,       // <4>
+                                       "live-after-8-seconds")          // <5>
```
Comment (spericas) [inline docs/se/health.adoc:133]: Method name has "Liveness" in it but type is READINESS?

File: docs/se/guides/health.adoc (hunk 1 above)
Comment (tjquinno) [reply docs/se/guides/health.adoc:200]: I retyped the source and our formatting template did trim some of the indentation. Fixed.

File: docs/se/guides/health.adoc (hunk 2 above)
Comment (tjquinno) [reply docs/se/guides/health.adoc:332]: Fixed.

File: docs/se/guides/health.adoc (hunk 4 above)
Comment (tjquinno) [reply docs/se/guides/health.adoc:385]: Fixed.

File: docs/se/health.adoc (hunk 6 above)
Comment (tjquinno) [reply docs/se/health.adoc:111]: Fixed.

File: docs/se/health.adoc (hunk 7 above)
Comment (tjquinno) [reply docs/se/health.adoc:133]: Good catch. I also reformatted this snippet to rein in the indentation a bit!

File: docs/se/guides/health.adoc (hunk 5 above)
Comment (tjquinno) [reply docs/se/guides/health.adoc:388]: Rephrased this a bit.

File: docs/se/guides/health.adoc (hunk 3 above)
Comment (tjquinno) [reply docs/se/guides/health.adoc:360]: I've rephrased this in a way that, I hope, is clearer.

**Review by spericas — APPROVED** [review]
